spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/render/OrderByClauseVisitor.java

```diff
@@ -77,7 +77,7 @@ Delegation leaveMatched(OrderByField segment) {
 	Delegation leaveNested(Visitable segment) {
 
 		if (segment instanceof Column) {
-			builder.append(NameRenderer.reference(context, (Column) segment));
+			builder.append(NameRenderer.fullyQualifiedReference(context, (Column) segment));
```

Comment:
Shouldn't this consider aliased column names as we want to order by a specific thing that is being selected (`SELECT foo.bar as fb, baz.bar as foo.bar … ORDER BY fb`)?